### ISSUE TITLE
Shift+Delete is not recorded properly

### ISSUE THREAD

**Issue by jakubklos77** [tshino/vscode-kb-macro#447]
When you use Shift+Delete instead of Ctrl+X (while recording a macro), it is not recorded and processed properly.
Is it possible to fix this?
Thank you

**Comment by tshino** [comment]
Thank you for reporting this issue.
However, I was not able to reproduce it.
I have tested it in a text editor tab. Recording Shift+Delete and its playback just worked for me.

Please let me know if you are facing this issue anywhere other than text editor tabs.
Also, this issue may happen when you are using a keymap extension or any other kind of extension that provides a keyboard shortcut of Shift+Delete or you have custom keybindings of Shift+Delete in your keybindings.json file. If so, please follow the instructions in the [README.md](https://github.com/tshino/vscode-kb-macro/blob/main/README.md).
Thanks.

**Comment by jakubklos77** [comment]
Thank you for your response.
No, no custom keybinding here and yes I test it in the text editor. Ctrl+X works but Shift+Delete (which should do exactly the same as Ctrl+X - cut to clipboard) does not work. I think the key is not recorded because when I play macro it does not even cut the text.

I am running on Linux Ubuntu 24.04. I am positive it will be another Gnome 46 bug for qt5 apps (there are so many bugs regarding Gnome). If it works for you then I guess nothing can be done on your part. At least I know a workaround.

Thank you

**Comment by tshino** [comment]
Thank you for more detailed information.
I am using VS Code on Windows 10 by the way.

To check whether the key has been recorded in the macro, you can use the command `Keyboard Macro: Copy Macro as Keybinding`.
If the key has not been recorded while the deletion was properly performed, then the keybinding wrapper for the key may be not working somehow I think.
To investigate it, you can use the [Developer: Toggle Keyboard Shortcuts Troubleshooting](https://code.visualstudio.com/docs/getstarted/keybindings#_troubleshooting-keybindings) command.

Here is the log where the keybinding wrapper (the command `kb-macro.wrap`) is properly invoked.
```
2024-12-09 23:29:48.425 [info] [KeybindingService]: / Received  keydown event - modifiers: [shift], code: ShiftLeft, keyCode: 16, key: Shift
2024-12-09 23:29:48.425 [info] [KeybindingService]: | Converted keydown event - modifiers: [shift], code: ShiftLeft, keyCode: 4 ('Shift')
2024-12-09 23:29:48.425 [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
2024-12-09 23:29:48.549 [info] [KeybindingService]: / Soft dispatching keyboard event
2024-12-09 23:29:48.550 [info] [KeybindingService]: | Resolving shift+Delete
2024-12-09 23:29:48.550 [info] [KeybindingService]: \ From 5 keybinding entries, matched kb-macro.wrap, when: kb-macro.active && kb-macro.headOfLine, source: user extension tshino.kb-macro.
2024-12-09 23:29:48.550 [info] [KeybindingService]: / Received  keydown event - modifiers: [shift], code: Delete, keyCode: 46, key: Delete
2024-12-09 23:29:48.550 [info] [KeybindingService]: | Converted keydown event - modifiers: [shift], code: Delete, keyCode: 20 ('Delete')
2024-12-09 23:29:48.550 [info] [KeybindingService]: | Resolving shift+Delete
2024-12-09 23:29:48.550 [info] [KeybindingService]: \ From 5 keybinding entries, matched kb-macro.wrap, when: kb-macro.active && kb-macro.headOfLine, source: user extension tshino.kb-macro.
2024-12-09 23:29:48.550 [info] [KeybindingService]: + Invoking command kb-macro.wrap.
2024-12-09 23:29:48.870 [info] [KeybindingService]: + Ignoring single modifier shift due to it being pressed together with other keys.
```

**Comment by jakubklos77** [comment]
Thank you
Yes, the shortcut was not recorded (it was not in the keybinding copy). The log below shows that kb-macro does not catch the Shift+Del:

```
2024-12-09 16:28:01.958 [info] [KeybindingService]: / Received  keydown event - modifiers: [shift], code: ArrowLeft, keyCode: 37, key: ArrowLeft
2024-12-09 16:28:01.959 [info] [KeybindingService]: | Converted keydown event - modifiers: [shift], code: ArrowLeft, keyCode: 15 ('LeftArrow')
2024-12-09 16:28:01.959 [info] [KeybindingService]: | Resolving shift+[ArrowLeft]
2024-12-09 16:28:01.960 [info] [KeybindingService]: \ From 4 keybinding entries, matched kb-macro.wrap, when: kb-macro.active && textInputFocus, source: user extension tshino.kb-macro.
2024-12-09 16:28:01.960 [info] [KeybindingService]: + Invoking command kb-macro.wrap.
2024-12-09 16:28:03.005 [info] [KeybindingService]: / Soft dispatching keyboard event
2024-12-09 16:28:03.005 [info] [KeybindingService]: | Resolving shift+[Delete]
2024-12-09 16:28:03.006 [info] [KeybindingService]: \ From 2 keybinding entries, no when clauses matched the context.
2024-12-09 16:28:03.006 [info] [KeybindingService]: / Received  keydown event - modifiers: [shift], code: Delete, keyCode: 46, key: Delete
2024-12-09 16:28:03.006 [info] [KeybindingService]: | Converted keydown event - modifiers: [shift], code: Delete, keyCode: 20 ('Delete')
2024-12-09 16:28:03.007 [info] [KeybindingService]: | Resolving shift+[Delete]
2024-12-09 16:28:03.007 [info] [KeybindingService]: \ From 2 keybinding entries, no when clauses matched the context.
2024-12-09 16:28:05.303 [info] [KeybindingService]: + Ignoring single modifier shift due to it being pressed together with other keys.
2024-12-09 16:28:06.596 [info] [KeybindingService]: / Soft dispatching keyboard event
2024-12-09 16:28:06.596 [info] [KeybindingService]: \ Keyboard event cannot be dispatched
2024-12-09 16:28:06.597 [info] [KeybindingService]: / Received  keydown event - modifiers: [ctrl], code: ControlLeft, keyCode: 17, key: Control
2024-12-09 16:28:06.597 [info] [KeybindingService]: | Converted keydown event - modifiers: [ctrl], code: ControlLeft, keyCode: 5 ('Ctrl')
2024-12-09 16:28:06.598 [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
2024-12-09 16:28:06.655 [info] [KeybindingService]: / Soft dispatching keyboard event
2024-12-09 16:28:06.656 [info] [KeybindingService]: \ Keyboard event cannot be dispatched
2024-12-09 16:28:06.656 [info] [KeybindingService]: / Received  keydown event - modifiers: [ctrl,alt], code: AltLeft, keyCode: 18, key: Alt
2024-12-09 16:28:06.656 [info] [KeybindingService]: | Converted keydown event - modifiers: [ctrl,alt], code: AltLeft, keyCode: 6 ('Alt')
2024-12-09 16:28:06.657 [info] [KeybindingService]: \ Keyboard event cannot be dispatched in keydown phase.
2024-12-09 16:28:07.020 [info] [KeybindingService]: / Soft dispatching keyboard event
2024-12-09 16:28:07.020 [info] [KeybindingService]: | Resolving ctrl+alt+[KeyR]
2024-12-09 16:28:07.021 [info] [KeybindingService]: \ From 8 keybinding entries, matched kb-macro.finishRecording, when: kb-macro.recording && config.keyboardMacro.recordingShortcuts == 'Option1', source: user extension tshino.kb-macro.
```

Does it help?
Thank you

**Comment by tshino** [comment]
Yes, it does!
Your log seems to be indicating that no keybindings corresponding to 'shift+delete'  were found.

I have searched 'shift+delete' in Keyboard Shortcuts on my VS Code and the result is following five entries:
![image](https://github.com/user-attachments/assets/fa8c4bea-4b01-478b-91fa-9129d0996bf8)

Would you please do the same thing and share the screenshot?

**Comment by tshino** [comment]
I found this vscode issue which may be related.
https://github.com/microsoft/vscode/issues/228175

**Comment by jakubklos77** [comment]
I have only these 3:

![Screenshot from 2024-12-10 07-33-35](https://github.com/user-attachments/assets/dfe735d3-48e4-449c-88a7-f7edae11d754)

Even if I add a new shortcut to the "Cut" command it still does not record that shortcut

**Comment by tshino** [comment]
Many thanks!
Apparently, the Linux version of VS Code does not handle and even define the key 'shift+delete' (other than for 'deleteFile'), but somehow the key works as a Cut operation when pressed on the editor.

I think the only we can do is work around.
Can you try adding this wrapper definition to your keybindings.json?
```
{
	"key": "shift+delete",
	"command": "kb-macro.wrap",
	"args": {
		"command": "editor.action.clipboardCutAction",
		"await": "document selection clipboard"
	},
	"when": "kb-macro.active && !kb-macro.headOfLine"
},
{
	"key": "shift+delete",
	"command": "kb-macro.wrap",
	"args": {
		"command": "editor.action.clipboardCutAction",
		"await": "document clipboard"
	},
	"when": "kb-macro.active && kb-macro.headOfLine"
},
```
If it works as expected, it allows you to record the key, and then I will be able to fix this issue in this way on my extension side.

**Comment by jakubklos77** [comment]
Wow, great, thank you. Yes, your way above makes the issue go away. So it is working now.
I checked twice (removed the binding - it was not working, added the binding - it was working again)
Thank you again

**Comment by tshino** [comment]
Nice!
Then I will add this work around to my extension.
I will close this issue after that.

**Comment by jakubklos77** [comment]
Awesome, thank you. Once released I will test it without the bindings and let you know. Thank you again!

**Comment by tshino** [comment]
I have just released v0.13.23 which includes the fix.
https://github.com/tshino/vscode-kb-macro/releases/tag/v0.13.23
Thank you so much!

**Comment by jakubklos77** [comment]
Works great! Thank you much :)!
